examples-gallery/plot_ball_rolling_on_spinning_disc.py

```diff
@@ -0,0 +1,620 @@
+T = sm.symbols('T', cls=sm.Function)
+Tdot, Tdotdot = sm.symbols('Tdot Tdotdot')
+
+mb, mo, g, r = sm.symbols('mb, mo, g, r')
+Omega, alpha = sm.symbols('Omega, alpha')
+
+N, A1, A2 = sm.symbols('N, A1 A2', cls=me.ReferenceFrame)
+O, CP, Dmc, m_Dmc = sm.symbols('O, CP, Dmc, m_Dmc', cls=me.Point)
+O.set_vel(N, 0)
+
+# %%
+# Determination of the holonomic constraints.
+#
+# If the ball rotates around the x axis by an angle q the
+# contact point will be move by -q1 * r in the A2.y direction.
+# If the ball rotates around the y axis by an angle q2 the
+# contact point will be move by q2 * r in the A2.x direction.
+# Hence the configuration constraints are:
+#
+# - x = r * q2
+# - y = -r * q1
+#
+# So the resulting speed constraints are:
+#
+# - d/dt(x) = r * d/dt(q2)
+# - d/dt(y) = -r * d/dt(q1)
+#
+#
+# The time t appears explicitly in the EOMs.
+# I copy the trick from JM in the example1.2.3 Parameter identification:
+# declare a function T(t), and then
+# pass the time as known trajectory. My OEMs also contain
+# d/dt(T(t)) and d^2/dx^2(T(t))
+# As T(t) = const * t I set these derivatives accordingly.
+
+# %%
+# Set up the equations of motion.
+# I do not use qdisc = sm.integrate(udisc, t), as this gives
+# a sm.Piecewise(..) result, likely not differentiable,
+# but use the result of this integration for alpha != 0.
+
+udisc = Omega * (1 - sm.exp(-alpha * T(t)))
+qdisc = (Omega * T(t) + Omega * sm.exp(-alpha * T(t)) / alpha -
+    Omega / alpha)
+A2.orient_axis(N, qdisc, N.z)
+A2.set_ang_vel(N, udisc*N.z)
+
+A1.orient_body_fixed(A2, (q1, q2, q3), '123')
+rot = A1.ang_vel_in(N)
+A1.set_ang_vel(A2, u1*A1.x + u2*A1.y + u3*A1.z)
+rot1 = A1.ang_vel_in(N)
+
+CP.set_pos(O, x*A2.x + y*A2.y)
+CP.set_vel(A2, ux*A2.x + uy*A2.y)
+
+Dmc.set_pos(CP, r*N.z)
+Dmc.set_vel(N, Dmc.pos_from(O).diff(t, N))
+
+m_Dmc.set_pos(Dmc, r*A1.x)
+m_Dmc.v2pt_theory(Dmc, N, A1)
+
+iXX = 2./5. * mb * r**2
+iYY = iXX
+iZZ = iXX
+
+I = me.inertia(A1, iXX, iYY, iZZ)
+ball = me.RigidBody('ball', Dmc, A1, mb, (I, Dmc))
+observer = me.Particle('observer', m_Dmc, mo)
+bodies = [ball, observer]
+
+forces = [(Dmc, -mb*g*N.z), (m_Dmc, -mo*g*N.z),
+    (A1, t1*A1.x + t2*A1.y + t3*A1.z)]
+
+kd = sm.Matrix([ux - x.diff(t), uy - y.diff(t),
+    *[(rot - rot1).dot(uv) for uv in N ]])
+speed_constr = sm.Matrix([ux - r*u2, uy + r*u1])
+hol_constr = sm.Matrix([x - r*q2, y + r*q1])
+
+q_ind = [q1, q2, q3]
+q_dep = [x, y]
+u_ind = [u1, u2, u3]
+u_dep = [ux, uy]
+
+KM = me.KanesMethod(N,
+    q_ind=q_ind,
+    q_dependent=q_dep,
+    u_ind=u_ind,
+    u_dependent=u_dep,
+    kd_eqs=kd,
+    velocity_constraints=speed_constr,
+    configuration_constraints=hol_constr,
+    )
+
+(fr, frstar) = KM.kanes_equations(bodies, forces)
+eom = kd.col_join(fr + frstar)
+eom = me.msubs((eom.col_join(hol_constr)),
+    {sm.Derivative(T(t), t):
+    Tdot, sm.Derivative(T(t), (t,2)): Tdotdot},
+    )
+print(f'eom contains {sm.count_ops(eom):,} operations, ' +
+    f'{sm.count_ops(sm.cse(eom))} after cse')
```

Comment:
count op not needed, remove

Reply:
> count op not needed, remove

I will remove. I put it there, as obviously these EOMs are too large to print them. I this would give the guy looking at the example an idea what EOMs he is using here.